### DISCUSSION TITLE
onPress

### DISCUSSION THREAD

**Discussion by jucian0** [adobe/react-spectrum#4748] [General]
I'm trying to implement a TagGroup component, but I got a warning and the remove feature does not work.
`react-dom.development.js:86 Warning: Unknown event handler property `onPress`. It will be ignored.
    at button`

To make it work, I replaced `onPress` to `onClick`

```jsx
    <TagLabel {...gridCellProps}>
        {item.rendered}
        {allowsRemoving && (
          <button
            {...removeButtonProps}
            onClick={(e) => removeButtonProps.onPress?.(e as any)}
          >
            <RemoveIcon />
          </button>
        )}
      </TagLabel>
```

Why do we don't have `onClick` property in `removeButtonProps`?

**Comment by reidbarber** [comment]
You'll want `removeButtonProps` to be spread onto a button that is made with `useButton` (since it takes `onPress`).

In your example, you could add 

```ts
  let { buttonProps } = useButton(removeButtonProps, ref);
```

then spread `buttonProps` onto the button, or create a re-usable button component like in the [docs example](https://react-spectrum.adobe.com/react-aria/useTagGroup.html#button).
